### PR TITLE
Fix BitArray.CopyTo to byte[] regression - update offset for extra bits.

### PR DESCRIPTION
**Pull request by ahsonkhan**
Fixes https://github.com/dotnet/core/issues/3194 and https://github.com/dotnet/corefx/issues/39929

We need to port this to release/3.0

cc @stephentoub, @jkotas, @safern, @MichaelL79, @carlossanlop, @Clockwork-Muse, @ericstj, @danielValdezR

### CODE REVIEW

File: src/System.Collections/src/System/Collections/BitArray.cs [hunk 1]
```diff
@@ -657,7 +657,7 @@ public void CopyTo(Array array, int index)
                     Debug.Assert(span.Length > 0);
                     Debug.Assert(m_array.Length > quotient);
                     // mask the final byte
-                    span[span.Length - 1] = (byte)((m_array[quotient] >> (remainder * 8)) & ((1 << (int)extraBits) - 1));
+                    span[remainder] = (byte)((m_array[quotient] >> (remainder * 8)) & ((1 << (int)extraBits) - 1));
```
Comment (ahsonkhan) [inline src/System.Collections/src/System/Collections/BitArray.cs]: Before the change from https://github.com/dotnet/corefx/pull/33367, this used to be as follows:
https://github.com/dotnet/corefx/blob/8716f1adc2f583d54b825998838e7cf7ed64cc0f/src/System.Collections/src/System/Collections/BitArray.cs#L545-L554

Before, we were writing one **byte** at a time, and then if the bit array contained extra **bits** to be copied over, we used the next index to write those bits. Now, we are writing **4 bytes** at a time, and if we have 1-3 bytes left to write, we have the switch on remainder below. Beyond that, any extra bits left to copy (i.e. < 1 byte) should go at offset `remainder`.

I believe using `remainder` as the offset to write the last byte to is correct (and the test pass). Please review/verify.

Comment (EgorBo) [reply src/System.Collections/src/System/Collections/BitArray.cs]: By the way
```csharp
(byte)((m_array[quotient] >> (remainder * 8)) & ((1 << (int)extraBits) - 1));
```
can be optimized in JIT: https://godbolt.org/z/XQ9Mpe
current codegen: [sharplab.io](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIGYACMhgYQYG8aaGeHgBPDDAZDcGABQBLAHYYG2NAxlzY+bDIAmMKIuUMYCDFGwAhSRlwBKDt14BIYgHYG4gUMvjx2BgD4fL1XVpLSgGACoGAA5LawAyF3FyBgAeZJdlSwMjU3MrBjgGchiAblseAF8acqA===)

Comment (ahsonkhan) [reply src/System.Collections/src/System/Collections/BitArray.cs]: cc @AndyAyersMS

File: src/System.Collections/tests/BitArray/BitArray_GetSetTests.cs [hunk 2]
```diff
@@ -312,6 +312,106 @@ public static void CopyTo<T>(BitArray bitArray, int length, int index, T[] expec
...
+
+                    expectedOutput[i] = 522;
+                    bitArray.CopyTo(actualOutput, i);
+
+                    Assert.Equal(expectedOutput, actualOutput);
+                }
+            }
+        }
+
+        // https://github.com/dotnet/corefx/issues/39929
```
Comment (safern) [inline src/System.Collections/tests/BitArray/BitArray_GetSetTests.cs]: Are these comments necessary?

Comment (safern) [reply src/System.Collections/tests/BitArray/BitArray_GetSetTests.cs]: I think the test name speaks for itself 😄

**Pull request merged**